probe/data.go

```diff
@@ -89,7 +95,9 @@ func CleanData(p []Prober) {
 			data[r.Name] = r
 		}
 	}
+	mutex.Lock()
```

Comment:
propose to put the whole function body into the lock (and avoid calling `GetResultData`), because `resultData` may contain old data if `SetResultData` was called after line 93.